x-pack/plugins/lens/public/indexpattern_plugin/indexpattern.test.tsx

```diff
@@ -188,13 +188,46 @@ describe('IndexPattern Data Source', () => {
 
             // Private
             operationType: 'value',
-            sourceField: 'op2',
+            sourceField: 'bytes',
           },
         },
       };
       const state = await indexPatternDatasource.initialize(queryPersistedState);
       expect(indexPatternDatasource.toExpression(state)).toMatchInlineSnapshot(
-        `"esdocs index=\\"1\\" fields=\\"op, op2\\" sort=\\"op, DESC\\""`
+        `"esdocs index=\\"1\\" fields=\\"source, bytes\\" sort=\\"source, DESC\\""`
+      );
+    });
+
+    it('should generate an expression for an aggregated query', async () => {
+      const queryPersistedState: IndexPatternPersistedState = {
+        currentIndexPatternId: '1',
+        columnOrder: ['col1', 'col2'],
+        columns: {
+          col1: {
+            operationId: 'op1',
+            label: 'Count of Documents',
+            dataType: 'number',
+            isBucketed: false,
+
+            // Private
+            operationType: 'count',
+            sourceField: 'document',
+          },
+          col2: {
+            operationId: 'op2',
+            label: 'Date',
+            dataType: 'date',
+            isBucketed: true,
+
+            // Private
+            operationType: 'date_histogram',
+            sourceField: 'timestamp',
+          },
+        },
+      };
+      const state = await indexPatternDatasource.initialize(queryPersistedState);
+      expect(indexPatternDatasource.toExpression(state)).toMatchInlineSnapshot(
+        `"esaggs index=\\"1\\" aggs=\\"{\\"timestamp\\":{\\"date_histogram\\":{\\"field\\":\\"timestamp\\",\\"aggs\\":{\\"document\\":{\\"count\\":{\\"field\\":\\"document\\"}}}}}}\\""`
       );
     });
   });
```

Comment:
Yes, I haven't added any test coverage so far because I wasn't sure about the expected functionality. Now that we've discussed I'll make some updates and add test coverage.